tests/testthat/test-scatter_plot.R

```diff
@@ -6,32 +6,82 @@ test_data_scatter2 <- data.frame(x = c("blue", "red", "yellow"), y = c(3,2,1))
   expect_error(scatter_plot(test_data_scatter, x, y, x, "y", "title", 10))
   expect_error(scatter_plot(test_data_scatter, x, y, "x", y, "title", 10))
   expect_error(scatter_plot(test_data_scatter, x, y, "x", "y", 10, 10))
   expect_error(scatter_plot(test_data_scatter, x, y, "x", "y", "title", "10"))
-  
+  expect_error(scatter_plot(test_data_scatter, x, y, "x", "y", "title", 10, "NULL"))
+  expect_error(scatter_plot(test_data_scatter, x, y, "x", "y", "title", 10, NULL))
+  expect_error(scatter_plot(test_data_scatter, x, y, "x", "y", "title", 10, "asdf"))
+  expect_error(scatter_plot(test_data_scatter, x, y, "x", "y", "title", 10, knn))
 })
 
-#test for output variables 
+
+
+
+#test for output variables for function with type=NULL
 testthat::test_that("Checks the output of the graph", {
-  plot <- scatter_plot(test_data_scatter, x, y, "x", "y", "Scatterplot Test", 10)
-  expect_true(ggplot2::is.ggplot(plot))
-  expect_equal(ggplot2::xlab(plot)$label, ggplot2::xlab(test_data_scatter)$label)
-  expect_equal(ggplot2::ylab(plot)$label, ggplot2::ylab(test_data_scatter)$label)
-  expect_equal(ggplot2::ggtitle(plot)$label, ggplot2::ggtitle(test_data_scatter)$label)
-  
+  plot1 <- scatter_plot(test_data_scatter, x, y, "x", "y", "Scatterplot Test", 10)
+  expect_true(ggplot2::is.ggplot(plot1))
+  expect_equal(ggplot2::xlab(plot1)$label, ggplot2::xlab(test_data_scatter)$label)
+  expect_equal(ggplot2::ylab(plot1)$label, ggplot2::ylab(test_data_scatter)$label)
+  expect_equal(ggplot2::ggtitle(plot1)$label, ggplot2::ggtitle(test_data_scatter)$label)
+  #expect_equal(ggplot2::type(plot1), type(test_data_scatter))
+})
+
+#test for output variables for function with type="lm"
+testthat::test_that("Checks the output of the graph", {
+  plot2 <- scatter_plot(test_data_scatter, x, y, "x", "y", "Scatterplot Test", 10)
+  expect_true(ggplot2::is.ggplot(plot2))
+  expect_equal(ggplot2::xlab(plot2)$label, ggplot2::xlab(test_data_scatter)$label)
+  expect_equal(ggplot2::ylab(plot2)$label, ggplot2::ylab(test_data_scatter)$label)
+  expect_equal(ggplot2::ggtitle(plot2)$label, ggplot2::ggtitle(test_data_scatter)$label)
+  #expect_equal(ggplot2::type(plot2), type=="lm")
+})
+
+#test for output variables for function with type="knn"
+testthat::test_that("Checks the output of the graph", {
+  plot3 <- scatter_plot(test_data_scatter, x, y, "x", "y", "Scatterplot Test", 10)
+  expect_true(ggplot2::is.ggplot(plot2))
+  expect_equal(ggplot2::xlab(plot2)$label, ggplot2::xlab(test_data_scatter)$label)
+  expect_equal(ggplot2::ylab(plot2)$label, ggplot2::ylab(test_data_scatter)$label)
+  expect_equal(ggplot2::ggtitle(plot2)$label, ggplot2::ggtitle(test_data_scatter)$label)
+  #expect_equal(ggplot2::type(plot2), type=="knn")
+})
+
+#test type output of scatter_plot
+testthat::test_that("Function has the right type", {
+  expect_type(helper_scatter_empty_type,'list')
+  expect_type(helper_scatter_lm_type ,'list')
+  expect_type(helper_scatter_knn_type ,'list')
```

Comment:
do `helper_scatter_empty_type`, `helper_scatter_lm_type` and `helper_scatter_knn_type` use `scatter_plot()` function? If not then they aren't testing the function.